SwiftCompilerSources/Sources/SIL/Instruction.swift

```diff
@@ -98,6 +98,19 @@ public class Instruction : ListNode, CustomStringConvertible, Hashable {
     }
   }
 
+  final public var mayReadRefCount: Bool {
```

Comment:
I thought about this again. The reason why we have a `SILInstruction::mayReleaseOrReadRefCount()` is because every release can call a destructor with arbitrary side effects, including reading the reference count.
E.g. instructions like `strong_release` or `apply` can read the refcount.
But fortunately this check is not needed in the release-devirtualizer: it's not legal SIL to read the ref count after the final release. So it's sufficient to check for `mayRelease`.
You can just remove `mayReadRefCount` here and in the release-devirtualizer.